### PR TITLE
[Update Sample] Updated app-auth sample and removed non working features

### PR DESCRIPTION
**Pull request by Prithvi-MSFT**

### CODE REVIEW

File: samples/app-auth/nodejs/.env [hunk 1]
```diff
@@ -0,0 +1,9 @@
+MicrosoftAppId=
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/.env:1]: Why we delete it?
image

File: samples/app-auth/nodejs/.env [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+MicrosoftAppPassword=
+TenantId=
+AppType=
+ConnectionName=
+FaceBookAppId=
+FaceBookAppPassword=
+FBConnectionName=
+ApplicationBaseUrl=
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/.env:9]: Check if you have already added steps in Readme for getting these values.

File: samples/app-auth/nodejs/README.md [hunk 3]
```diff
@@ -80,8 +80,7 @@ To be able to use an identity provider, first you have to register your applicat
 
 This project uses the [config](https://www.npmjs.com/package/config) package. The default configuration is in `config\default.json`.
 
--   Environment variable overrides are defined in `config\custom-environment-variables.json`. You can set these environment variables when running node. If you are using Visual Studio Code, you can set these in your `launch.json` file.
--   Alternatively, you can specify local modifications in `config\local.json`.
+-   Environment variable overrides are defined in `.env`. You can set these environment variables when running node.
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/README.md:83]: Where to update and where .env file will be there. add path

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:1]: We don't need this file?
image

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    constructor(id) {
+        super(id, process.env.connectionName);
+
+        this.addDialog(new SsoOAuthPrompt(OAUTH_PROMPT, {
+            connectionName: process.env.connectionName,
+            text: 'Please Sign In',
+            title: 'Sign In',
+            timeout: 300000
+        }));
+        this.addDialog(new TextPrompt(TEXT_PROMPT));
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:25]: One line space before it.

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        this.addDialog(new WaterfallDialog(SSOAUTH, [
+            this.promptStep.bind(this),
+            this.loginStep.bind(this),
+            this.userInfoStep.bind(this)
+        ]));
+        this.initialDialogId = SSOAUTH;
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:31]: One line space before it.

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+                await stepContext.context.sendActivity('Login was not successful please try again.');
+            }
+            else {
+                token = tokenResponse.token;
+                const messageText = 'What is your user name?';
+                const msg = MessageFactory.text(messageText, messageText, InputHints.ExpectingInput);
+                await stepContext.context.sendActivity('Login successful.');
+                return await stepContext.prompt(TEXT_PROMPT, { prompt: msg });
+            }
+            await stepContext.context.sendActivity("Please type 'sso' to begin authentication");
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:54]: One line space before it.

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+            await userImage.arrayBuffer().then(result => {
+                console.log(userImage.type);
+                imageString = Buffer.from(result).toString('base64');
+                img2 = "data:image/png;base64," + imageString;
+            }).catch(error => { console.log(error) });
+
+            const userCard = CardFactory.adaptiveCard(this.getAdaptiveCardUserDetails(myDetails, img2,userName));
+            await stepContext.context.sendActivity({ attachments: [userCard] });
+        }
+        return await stepContext.endDialog();
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:78]: One line space before it.

File: samples/app-auth/nodejs/dialogs/mainDialog.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+                return await stepContext.beginDialog(FACEBOOKAUTH);
+            }
+            else{
+                const buttons = [
+                    { type: ActionTypes.ImBack, title: 'AAD SSO authentication', value: 'sso' },
+                    { type: ActionTypes.ImBack, title: 'Facebook login (OAuth 2)', value: 'facebooklogin' }]
+                const card = CardFactory.heroCard('Login options', undefined,
+                    buttons,{"text":"Select a login option"});
+                await stepContext.context.sendActivity({ attachments: [card] });
+                return await stepContext.endDialog();
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/mainDialog.js:60]: One line space before it.

File: samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    constructor(id) {
+        super(id, process.env.FBConnectionName);
+
+        this.addDialog(new OAuthPrompt(OAUTH_PROMPT, {
+            connectionName: process.env.FBConnectionName,
+            text: 'Login to facebook',
+            title: 'Log In',
+            timeout: 300000
+        }));
+        this.addDialog(new TextPrompt(TEXT_PROMPT));
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:24]: One line space before it.

File: samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+        }
+        
+        await stepContext.context.sendActivity('Login was not successful please try again.');
+        return await stepContext.endDialog();
+    }
+
+ async userInfoStep(stepContext) {
+   const userName = stepContext.result
+   var facbookProfile = await this.getFacebookUserData(token);
+        const profileCard = CardFactory.adaptiveCard({
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:57]: One line space before it.

File: samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+                    type: 'TextBlock',
+                    text: 'Hello: ' + facbookProfile.name,
+                },
+                {
+                    type: 'TextBlock',
+                    text: 'User name ' + userName,
+                },
+            ],
+        });
+    await stepContext.context.sendActivity({attachments:[profileCard]});
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:76]: One line space before it.

File: samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+    async getFacebookUserData(access_token) {
+        const { data } = await axios({
+            url: 'https://graph.facebook.com/v2.6/me',
+            method: 'get',
+            params: {
+                fields: ['name','picture'].join(','),
+                access_token: access_token,
+            },
+        });
+        return data;
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:90]: One line space before it.

Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:90]: What is data?

Add proper naming conventions.

File: samples/app-auth/nodejs/index.js [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+  var oboPromise = new Promise((resolve, reject) => {
+    msalClient.acquireTokenOnBehalfOf({
+      authority: `https://login.microsoftonline.com/${tid}`,
+      oboAssertion: token,
+      scopes: scopes,
+      skipCache: true
+    }).then(async result => {
+      const client = new SimpleGraphClient(result.accessToken);
+      const myDetails = await client.getMeAsync();
+      try {
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/index.js:159]: One line space before it.

File: samples/app-auth/nodejs/index.js [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+    res.json(err);
+  });
+});
+
+// Listen for incoming requests.
+server.post('/decodedToken', async (req, res) => {
+  var idToken = req.body.idToken;
+  const decodedToken = jwt.decode(idToken, { complete: true });
+    var responseMessage = Promise.resolve(decodedToken.payload);
+    responseMessage.then(function (result) {
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/index.js:194]: One line space before it.

File: samples/app-auth/nodejs/views/auth-start.ejs [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.2/js/bootstrap.min.js"></script>
+     <script type="text/javascript">
+        microsoftTeams.app.initialize().then(() => {
+            // Get the tab context, and use the information to navigate to Azure AD login page
+            microsoftTeams.app.getContext().then(async (context) => {
+                var currentURL = new URL(window.location);
+                var scope = "User.Read email openid profile offline_access";
+                var loginHint = context.user.loginHint;
+                const msalConfig = {
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/views/auth-start.ejs:13]: One line space before it.

File: samples/app-auth/nodejs/views/silent-tab.ejs [hunk 17]
```diff
@@ -0,0 +1,265 @@
...
+            }
+        };
+        const msalInstance = new msal.PublicClientApplication(msalConfig);
+
+        microsoftTeams.app.initialize().then(() => {
+            microsoftTeams.app.getContext().then((context) => {
+                upn = context.user.userPrincipalName;
+                let cachedAccounts = msalInstance.getAllAccounts();
+                let accountInfo = cachedAccounts.filter(account => account.username === upn);
+                if (accountInfo.length > 0) {
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/views/silent-tab.ejs:71]: One line space before it.

File: samples/app-auth/nodejs/views/silent-tab.ejs [hunk 18]
```diff
@@ -0,0 +1,265 @@
...
+                            // At this point we have to get the user involved, so show the login button
+                            $("#btnLogin").show();
+                        });
+                } else {
+                    $("#btnLogin").show();
+                }
+            });
+        });
+
+
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/views/silent-tab.ejs:99]: Remove extra empty lines.

File: samples/app-auth/nodejs/views/silent-tab.ejs [hunk 19]
```diff
@@ -0,0 +1,265 @@
...
+            // Parse query parameters into key-value pairs
+            function getQueryParameters() {
+                let queryParams = {};
+                location.search.substr(1).split("&").forEach(function (item) {
+                    let s = item.split("="),
+                        k = s[0],
+                        v = s[1] && decodeURIComponent(s[1]);
+                    queryParams[k] = v;
+                });
+                return queryParams;
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/views/silent-tab.ejs:246]: One line space before it.

File: samples/app-auth/nodejs/views/silent-tab.ejs [hunk 20]
```diff
@@ -0,0 +1,265 @@
...
+            }
+
+            // Demonstrates silent logout - simply clears the loginHint, replaces the value of userObjectId with a dummy value, and reloads the page
+            function logout() {
+                localStorage.clear();
+                let url = location.href.split("?")[0] + "?";
+                let queryParams = getQueryParameters();
+                delete queryParams["loginHint"];
+                queryParams["userObjectId"] = "00000000-0000-0000-000000000000";
+                for (var k in queryParams) {
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/views/silent-tab.ejs:256]: One line space before it.

File: samples/app-auth/nodejs/views/silent-tab.ejs [hunk 21]
```diff
@@ -0,0 +1,265 @@
...
+                    url = url + k + "=" + encodeURIComponent(queryParams[k]) + "&";
+                    console.log(k);
+                }
+                location.href = url;
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/views/silent-tab.ejs:260]: One line space before it.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my review comments.

File: samples/app-auth/nodejs/README.md [hunk 22]
```diff
@@ -184,44 +181,6 @@ Notes:
 5. Click **Save**.
 6. Set the environment variable `AZUREAD_CONNECTIONNAME` to the name that you chose for this OAuth connection.
 
-### [Optional] Using LinkedIn
-
```
[the remark this answers is not in the document]
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/README.md:188]: Updated with facebook auth

File: samples/app-auth/nodejs/appPackage/manifest.json [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+  "version": "1.0.0",
+  "id": "{{Microsoft-App-Id}}",
+  "packageName": "com.microsoft.teams.appauth",
+  "webApplicationInfo": {
+    "id": "{{Microsoft-App-Id}}",
+    "resource": "api://{Base_URL_Domain}/botid-{{Microsoft-App-Id}}"
+  },
+  "developer": {
+    "name": "Microsoft",
+    "websiteUrl": "https://example.azurewebsites.net",
```
[the remark this answers is not in the document]
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/appPackage/manifest.json:13]: This sample is yet to be migrated for toolkit

File: samples/app-auth/nodejs/.env (hunk 1 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/.env:1]: Not required, we will add toolkit scripts in another PR

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js (hunk 4 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:1]: Yes not needed

File: samples/app-auth/nodejs/views/silent-tab.ejs (hunk 21 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/views/silent-tab.ejs:260]: added

File: samples/app-auth/nodejs/views/silent-tab.ejs (hunk 19 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/views/silent-tab.ejs:246]: added

File: samples/app-auth/nodejs/views/silent-tab.ejs (hunk 18 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/views/silent-tab.ejs:99]: removed

File: samples/app-auth/nodejs/views/silent-tab.ejs (hunk 17 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/views/silent-tab.ejs:71]: added

File: samples/app-auth/nodejs/views/auth-start.ejs (hunk 16 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/views/auth-start.ejs:13]: added

File: samples/app-auth/nodejs/index.js (hunk 15 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/index.js:194]: added

File: samples/app-auth/nodejs/index.js (hunk 14 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/index.js:159]: added

File: samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js (hunk 13 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:90]: The API itself returns with key as "data" so we need to keep the same variable name

Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:90]: added

File: samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js (hunk 12 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:76]: added

File: samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js (hunk 11 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:57]: added

File: samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js (hunk 10 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:24]: added

File: samples/app-auth/nodejs/dialogs/mainDialog.js (hunk 9 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/mainDialog.js:60]: added

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js (hunk 5 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:25]: added

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js (hunk 6 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:31]: added

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js (hunk 7 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:54]: added

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js (hunk 8 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:78]: added

File: samples/app-auth/nodejs/bots/teamsBot.js [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+            await next();
+        });
+    }
+
+    async handleTeamsSigninTokenExchange(context, query) {
+        console.log('Running dialog with signin/tokenExchange from an Invoke Activity.');
+        await this.dialog.run(context, this.dialogState);
+    }
+
+
```
[the remark this answers is not in the document]
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/bots/teamsBot.js:40]: added

File: samples/app-auth/nodejs/assets/sample.json [hunk 25]
```diff
@@ -1,15 +1,15 @@
...
+    "shortDescription": "This sample demos Azure AD and facebook authentication feature in bot,tab and messaging extension.",
+    "url": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-complete-auth/nodejs",
     "longDescription": [
-      "This sample demonstrates authentication in Microsoft Teams using bot and tab."
+      "This sample demos Azure AD and facebook authentication feature in bot,tab and messaging extension."
     ],
-    "creationDateTime": "2018-02-08",
-    "updateDateTime": "2022-06-20",
+    "creationDateTime": "2022-02-16",
+    "updateDateTime": "2022-05-17",
```
[the remark this answers is not in the document]
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/assets/sample.json:12]: changed

File: samples/app-auth/nodejs/README.md [hunk 26]
```diff
@@ -184,44 +187,6 @@ Notes:
...
-1. Obtain OAuth2 client credentials from the [Google API Console](https://console.developers.google.com). Enable access to the [Google People API](https://developers.google.com/people/).
-2. In "Authorized redirect URLs", add `https://token.botframework.com/.auth/web/redirect`.
-3. Note your app's "Client ID" and "Client Secret".
-4. Navigate to your Azure bot's service page on the [Azure Portal](https://ms.portal.azure.com/#blade/HubsExtension/BrowseResourceBlade/resourceType/Microsoft.BotService%2FbotServices).
-5. Click **Settings**.
-6. Under **OAuth Connection Settings** near the bottom of the page, click **Add Setting**.
-7. Fill in the form as follows:
-
-    1. For **Name**, enter a name for your connection (e.g., "Google")
-    2. For **Service Provider**, select **Google**.
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/README.md:217]: You removed linked in related auth, so please make sure that images/gif are also updated.

File: samples/app-auth/nodejs/assets/sample.json [hunk 27]
```diff
@@ -1,15 +1,15 @@
...
+    "title": "Authentication complete sample",
+    "shortDescription": "This sample demos Azure AD and facebook authentication feature in bot,tab and messaging extension.",
+    "url": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/app-complete-auth/nodejs",
     "longDescription": [
-      "This sample demonstrates authentication in Microsoft Teams using bot and tab."
+      "This sample demos Azure AD and facebook authentication feature in bot,tab and messaging extension."
     ],
-    "creationDateTime": "2018-02-08",
-    "updateDateTime": "2022-06-20",
+    "creationDateTime": "2022-02-16",
```
Comment (ChetanSharma-msft) [inline samples/app-auth/nodejs/assets/sample.json:11]: Do we need to update created date as well?

File: samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js (hunk 4 above)
Comment (ChetanSharma-msft) [reply samples/app-auth/nodejs/dialogs/botSSOAuthDialog.js:1]: ok

File: samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js (hunk 13 above)
Comment (ChetanSharma-msft) [reply samples/app-auth/nodejs/dialogs/simpleFacebookAuthDialog.js:94]: ok

**Review by ChetanSharma-msft — APPROVED** [review]
Approving the PR with few suggestions, please check and fix those.

File: samples/app-auth/nodejs/assets/sample.json (hunk 27 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/assets/sample.json:11]: We can change updated date only

File: samples/app-auth/nodejs/README.md (hunk 26 above)
Comment (Prithvi-MSFT) [reply samples/app-auth/nodejs/README.md:217]: updated images and gif

**Review by Pawank-MSFT — APPROVED** [review]
looks correct, approving it!

**Review by Wajeed-msft — APPROVED** [review]

**Comment by Prithvi-MSFT** [comment]
Build status - All builds passed
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6350982950

**Pull request merged**